### PR TITLE
Update go to version 1.18.1 and terraform to version 1.0.1

### PR DESCRIPTION
**Pull request by kimsu98**
### What changes are proposed in this pull request?

Upgrade docker images to use more recent go version 1.18.1 and terraform 1.0.1

### Why are the changes needed?

Needed to remain consistent with version standards for tooling

### Does this PR introduce any user facing changes?

Indirectly user-facing with new docker image versions to be released in docker hub.

### CODE REVIEW

File: dev/jenkins/Dockerfile-jdk11 [hunk 1]
```diff
@@ -16,7 +16,7 @@ FROM maven:3.6.3-jdk-11
 # reference: https://github.com/docker-library/openjdk/blob/master/8/jdk/buster/Dockerfile 
 # we need jdk 8 in jdk 11 so that we can compile with jdk 8 and test with jdk 11
 
-ENV JAVA_HOME /usr/local/openjdk-8
+ENV JAVA_HOME /usr/local/openjdk-11
```
Comment (Xenorith) [inline dev/jenkins/Dockerfile-jdk11:19]: what is this change for? it should stay as java 8 at this point, see the above comment

Comment (kimsu98) [reply dev/jenkins/Dockerfile-jdk11]: Attempt to build image locally failed and thought this was the cause but it was unrelated. Reverted this change and am able to build locally fine.

**Review by Xenorith — APPROVED** [review]

**Comment by jiacheliu3** [comment]
@Xenorith Let's merge this if it looks okay?

**Comment by Xenorith** [comment]
PR is out of date, versions updated as part of https://github.com/Alluxio/alluxio/pull/15881

**Pull request closed without merging**